src/arrayMethodSort.js

```diff
@@ -4,8 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    const compareFn =
+      compareFunction ||
+      function (aValue, bValue) {
+        const a = String(aValue);
+        const b = String(bValue);
+
+        return a < b ? -1 : a > b ? 1 : 0;
+      };
+
+    for (let i = 0; i < this.length - 1; i++) {
+      for (let j = 0; j < this.length - 1 - i; j++) {
+        if (compareFn(this[j], this[j + 1]) > 0) {
```

Comment:
Using a temporary variable `temp` for swapping elements is a valid approach. Alternatively, you can also use array destructuring to swap the elements, which can make the code slightly more concise: `[this[j], this[j + 1]] = [this[j + 1], this[j]];`